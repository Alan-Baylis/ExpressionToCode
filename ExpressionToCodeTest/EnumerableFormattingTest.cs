using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using ExpressionToCodeLib;
using Xunit;

namespace ExpressionToCodeTest
{
    public class EnumerableFormattingTest
    {
        [Fact]
        public void ShortArraysRenderedCompletely()
        {
            Assert.Equal("new[] {1, 2, 3}", ObjectToCode.ComplexObjectToPseudoCode(new[] { 1, 2, 3 }));
        }

        [Fact]
        public void ShortEnumerablesRenderedCompletely()
        {
            Assert.Equal("{1, 2, 3}", ObjectToCode.ComplexObjectToPseudoCode(Enumerable.Range(1, 3)));
        }

        [Fact]
        public void LongEnumerablesBreakAfter10()
        {
            Assert.Equal("{1, 2, 3, 4, 5, 6, 7, 8, 9, 10, ...}", ObjectToCode.ComplexObjectToPseudoCode(Enumerable.Range(1, 13)));
        }

        [Fact]
        public void LongArraysBreakAfter10()
        {
            Assert.Equal("new[] {1, 2, 3, 4, 5, 6, 7, 8, 9, 10, ...}", ObjectToCode.ComplexObjectToPseudoCode(Enumerable.Range(1, 13).ToArray()));
        }
    }
}
