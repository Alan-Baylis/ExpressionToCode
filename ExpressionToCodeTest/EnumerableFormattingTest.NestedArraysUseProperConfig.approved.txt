new[] {
    null,
    new {
        A = 3,
        B = new[] {1, 2, 3, ...},
    ...,
}